src/disputeManager/DisputeManager.ts

```diff
@@ -127,6 +131,7 @@ class DisputeManager {
                 }
             }
 
+            this.storage.disputes.storeDispute(dispute);
```

Comment:
This should be stored when the event is observed, since here you have no gurantee that the dispute would succeed and if you do it like this you have to pair it with `delete` in the case of an error as I did with `this.storage.disputes.storeDisputedFork(forkId, true)` and `this.storage.disputes.storeDisputedFork(forkId, false)` bellow, but I think it's simpler to just store it when you observe it since the pipeline is clean and simple